src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -361,27 +361,52 @@ case class JesBackend(actorSystem: ActorSystem)
     executeOrResume(backendCall, runIdForResumption = runId)
   }
 
-  def useCachedCall(cachedCall: BackendCall, backendCall: BackendCall)(implicit ec: ExecutionContext): Future[ExecutionHandle] = Future {
-    val log = workflowLoggerWithCall(backendCall)
-    authenticateAsUser(backendCall.workflowDescriptor) { interface =>
-      Try(interface.copy(cachedCall.callGcsPath.toString, backendCall.callGcsPath.toString)) match {
-        case Failure(ex) =>
-          log.error(s"Exception occurred while attempting to copy outputs from ${cachedCall.callGcsPath} to ${backendCall.callGcsPath}", ex)
-          FailedExecutionHandle(ex).future
-        case Success(_) => postProcess(backendCall) match {
-          case Success(outputs) => backendCall.hash map { h =>
-            SuccessfulExecutionHandle(outputs, Seq.empty[ExecutionEventEntry], backendCall.downloadRcFile.get.stripLineEnd.toInt, h, Option(cachedCall)) }
-          case Failure(ex: AggregatedException) if ex.exceptions collectFirst { case s: SocketTimeoutException => s } isDefined =>
-            // TODO: What can we return here to retry this operation?
-            // TODO: This match clause is similar to handleSuccess(), though it's subtly different for this specific case
-            val error = "Socket timeout occurred in evaluating one or more of the output expressions"
-            log.error(error, ex)
-            FailedExecutionHandle(new Throwable(error, ex)).future
-          case Failure(ex) => FailedExecutionHandle(ex).future
-        }
+  def useCachedCall(cachedCall: BackendCall, backendCall: BackendCall)(implicit ec: ExecutionContext): Future[ExecutionHandle] = {
+    import better.files._
+
+    def renameCallSpecificFiles = {
+      val namesBuilder = List(
+        JesBackendCall.jesLogStdoutFilename _,
+        JesBackendCall.jesLogStderrFilename _,
+        JesBackendCall.jesLogFilename _,
+        JesBackendCall.jesReturnCodeFilename _
+      )
+
+      val wrongPaths = namesBuilder map { _(cachedCall.key) } map backendCall.callGcsPath.resolve
+      val correctNames = namesBuilder map { _(backendCall.key) }
```

Comment:
@cjllanwarne  Yep - Because  the log, stdout, stderr and return code files have names specific to the call/shard, call caching breaks when trying to read those files because they get copied as is, with their original name from the to cached call, and the original does not necessarily match the name expected by the the current call. I'll put this in a comment.
We might want to discuss if it's really necessary to have the name of the call + shard ( + attempt that should be there too probably) in the name of those files.